lib/astutils.cpp

```diff
@@ -2419,9 +2419,15 @@ static bool isExpressionChangedAt(const F& getExprTok,
     if (depth < 0)
         return true;
     if (tok->exprId() != exprid) {
-        if (globalvar && !tok->isKeyword() && Token::Match(tok, "%name% (") && !(tok->function() && tok->function()->isAttributePure()))
-            // TODO: Is global variable really changed by function call?
-            return true;
+        if (globalvar && !tok->isKeyword() && Token::Match(tok, "%name% (") && !(tok->function() && tok->function()->isAttributePure())) {
+            const Token* exprTok = getExprTok();
+            if (!((exprTok && exprTok->variable() && exprTok->variable()->scope() && tok->function() && tok->function()->isConst() &&
+                   (exprTok->variable()->scope() == tok->function()->nestedIn || // member variable doesn't get changed within const member function
+                    exprTok->variable()->typeScope() == tok->function()->nestedIn)) || // variable doesn't get changed by calling const member function on it
```

Comment:
A global variable could still be modified even if it is passed as a const parameter.

Reply:
I should probably call `isClassOrStruct()` on the scope, and also check for static variables.